compositor_render/src/transformations/shader/validation.rs

```diff
@@ -637,75 +604,99 @@ impl TypeInnerExt for naga::TypeInner {
                 class,
             } => match (dim, arrayed, class) {
                 (naga::ImageDimension::D1, false, naga::ImageClass::Sampled { kind, .. }) => {
-                    format!("texture_1d<{}>", kind.to_string(4))
```

Comment:
By removing ScalarKindExt this function becomes far bigger than it needs to be (especially when just a single case statement generates that much code). Can you rework it to be more readable?